### ISSUE TITLE
Apps/libraries/sysimages that have no package dependancies trigger unnecessary warning

### ISSUE THREAD

**Issue by LilithHafner** [JuliaLang/PackageCompiler.jl#692]
If I make a simple package with no dependencies and compile it to a sysimage/app/library with PackageCompiler, I get:
```
┌ Warning: it is not recommended to create an app/library without a preexisting manifest
└ @ PackageCompiler ~/.julia/packages/PackageCompiler/X7R0D/src/PackageCompiler.jl:66
```
Even though I do have a preexisting manifest file, it's just empty:
```
# This file is machine-generated - editing it directly is not advised

julia_version = "1.7.3"
manifest_format = "2.0"

[deps]

```
This is because a manifest without dependencies is interpreted as a missing manifest.
https://github.com/JuliaLang/PackageCompiler.jl/blob/a8a4459e7bd8c5dc2194aa62e93592e416edbdfa/src/PackageCompiler.jl#L65

([from discourse](https://discourse.julialang.org/t/packagecompiler-jl-warning-it-is-not-recommended-to-create-an-app-library-without-a-preexisting-manifest/81942))

**Comment by DilumAluthge** [comment]
I believe this is closed by the combination of https://github.com/JuliaLang/PackageCompiler.jl/pull/705 and https://github.com/JuliaLang/PackageCompiler.jl/pull/715?
